include/dxc/Support/Path.h

```diff
@@ -96,28 +114,16 @@ StringTy NormalizePathImpl(const CharT *Path, size_t Length) {
       PathCopy[i] = SlashTo;
   }
 
-  // Remove double slashes.
-  bool SeenNonSlash = false;
-  for (unsigned i = 0; i < PathCopy.size();) {
-    // Remove this slash if:
-    // 1. It is preceded by another slash.
-    // 2. It is NOT part of a series of leading slashes. (E.G. \\, which on
-    // windows is a network path).
-    if (PathCopy[i] == SlashTo && i > 0 && PathCopy[i - 1] == SlashTo &&
-        SeenNonSlash) {
-      PathCopy.erase(PathCopy.begin() + i);
-      continue;
-    }
-    SeenNonSlash |= PathCopy[i] != SlashTo;
-    i++;
-  }
+  RemoveDoubleSlashes<CharT, StringTy>(PathCopy, SlashTo);
 
   // If relative path, prefix with dot.
   if (IsAbsoluteOrCurDirRelativeImpl<CharT>(PathCopy.c_str(),
                                             PathCopy.size())) {
     return PathCopy;
   } else {
-    return StringTy(1, CharT('.')) + StringTy(1, SlashTo) + PathCopy;
+    PathCopy = StringTy(1, CharT('.')) + StringTy(1, SlashTo) + PathCopy;
+    RemoveDoubleSlashes<CharT, StringTy>(PathCopy, SlashTo);
+    return PathCopy;
```

Comment:
It really feels like all of this should be available in the llvm::sys::path utilities, but I sure can't find it 😕

Reply:
There's `llvm::sys::path::native`. However `native` replaces all double backslash with single backslash, which is not really what we want for UNC paths. The `RemoveDoubleSlashes` functionality is definitely not in there. Even if it is, I would like to keep this functionality header only to avoid linking problems.